src/test/ValidatorKeys_test.cpp

```diff
@@ -259,6 +260,29 @@ class ValidatorKeys_test : public beast::unit_test::suite
     }
 
     void
+    testSign ()
+    {
+        testcase ("Sign");
+
+        std::string const data = "data to sign";
+
+        for (auto const keyType : keyTypes)
+        {
+            ValidatorKeys keys (keyType);
+
+            auto const signature = keys.sign (data);
+
+            auto const ret = strUnHex (signature);
+            BEAST_EXPECT (ret.second);
+            BEAST_EXPECT (ret.first.size ());
+            BEAST_EXPECT (verify (
+                keys.publicKey(),
+                makeSlice (data),
+                makeSlice (ret.first)));
+        }
+    }
```

Comment:
IMHO, this test could be improved by making it deterministic and checking the signature. If you agree: https://github.com/ximinez/validator-keys-tool/commit/90fa0f37130c3dd410cc7b3a6e9895ef3a3a02d6